src/web/core/src/main/java/org/geoserver/web/wicket/HTTPURLsListTextArea.java

```diff
@@ -0,0 +1,101 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.web.wicket;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
+import org.apache.wicket.markup.html.form.TextArea;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.util.convert.IConverter;
+import org.apache.wicket.validation.IValidatable;
+import org.apache.wicket.validation.IValidationError;
+import org.apache.wicket.validation.IValidator;
+import org.apache.wicket.validation.ValidationError;
+
+/** TextArea for specifying a list of HTTP Urls (separated by new lines). */
+public class HTTPURLsListTextArea extends TextArea<List<String>> {
+
+    private static final long serialVersionUID = -8195179437229644665L;
+
+    public HTTPURLsListTextArea(String id, IModel<List<String>> model) {
+        super(id, model);
+
+        add(new HTTPURLsListValidator());
+        setType(List.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <C> IConverter<C> getConverter(Class<C> type) {
+        if (List.class.isAssignableFrom(type)) {
+            return (IConverter<C>) new URLsListConverter();
+        }
+        return super.getConverter(type);
+    }
+
+    private static class URLsListConverter implements IConverter<List<String>> {
+        private static final long serialVersionUID = 1083795866666107798L;
+
+        static final Pattern NEW_LINE_SEPARATED =
+                Pattern.compile("\\s*\\r?\\n\\s*", Pattern.MULTILINE);
```

Comment:
Made me wonder about just "\r", but looks like OSX always used "\n" instead, and classic MacOS is long gone.